include/albatross/src/core/distribution.hpp

```diff
@@ -165,8 +176,17 @@ struct JointDistribution : public DistributionBase<JointDistribution> {
     return covariance.diagonal()[i];
   }
 
+  bool approximately_equal(
+      const JointDistribution &other,
+      const double epsilon = cDefaultApproximatelyEqualEpsilon) const {
+    const bool mean_approx_equal = mean.isApprox(other.mean, epsilon);
+    const bool cov_approx_equal =
+        covariance.diagonal().isApprox(other.covariance.diagonal(), epsilon);
```

Comment:
I think we should drop the `diagonal()` here, so it's just:
```
covariance.isApprox(other.covariance, epsilon);
```

Reply:
Yeah. just saw the Joint type could be populated everywhere in the cov matrix